source/common/umf_pools/disjoint_pool.cpp

```diff
@@ -293,6 +293,16 @@ class DisjointPool::AllocImpl {
         // Generate buckets sized such as: 64, 96, 128, 192, ..., CutOff.
         // Powers of 2 and the value halfway between the powers of 2.
         auto Size1 = params.MinBucketSize;
+        // Special case for when MinBucketSize equals 1 because otherwise bucket sizes starting from 1 value
+        // would generate with duplicates (like 1,1,2,2,4,4...) which is undesirable.
+        // In this case buckets are generated sized such as: 1, 2, 4, 8, 16, 32, 64, 96, 128 ..., Cutoff.
+        // Powers of 2 to the size of 64 and then powers of 2 and the values halfway
+        // between the powers of 2.
+        if (Size1 == 1) {
```

Comment:
What if `Size1` is greater than `1`? I think the `if` statement is excessive because we do not need half the power of `2` until `64`.

Reply:
Another question: does it make sense to have such small bucket sizes? Maybe we can just round up the `Size1` to `64` or `32`?

Reply:
I don't think we have a use case for such small buckets but since this option is exposed why should we artificially limit it to 64 or 32?

Reply:
I would agree that having buckets for anything below 8 bytes is excessive, but it shouldn't be much of an issue (unless it causes some alignment issues or false sharing).

However, 1 appears to be the default value, so we probably shouldn't change it, at least not without investigating the impact.

Reply:
As for the MinBucketSize default value, It was set to 0 before but I've changed it to 1 in the previous PR (since 0 caused an infinite loop in the sizes generation and also the MinBucketSize was ensured to be a power of two in that PR). So I think it should be safe to change to another value if needed.

Reply:
I changed the MinBucketSize default value to 8 and also set 8 to minimal value from which sizes for buckets are generated.